src_c/music.c

```diff
@@ -496,6 +496,66 @@ music_queue(PyObject *self, PyObject *args, PyObject *keywds)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+music_get_metadata(PyObject *self, PyObject *args, PyObject *keywds)
+{
+    PyObject *meta_dict;
+    Mix_Music *music = current_music;
+
+    PyObject *obj = NULL;
+    char *namehint = NULL;
+    static char *kwids[] = {"filename", "namehint", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, keywds, "|Os", kwids, &obj,
+                                     &namehint))
+        return NULL;
+
+    MIXER_INIT_CHECK();
+
+    if (obj) {
+        music = _load_music(obj, namehint);
+        if (!music) {
+            return NULL;
+        }
+    }
+    else if (namehint) {
+        PyErr_SetString(
+            PyExc_RuntimeError,
+            "'namehint' specified without specifying 'filename' or 'fileobj'");
+        return NULL;
+    }
+
+    const char *title = "";
+    const char *album = "";
+    const char *artist = "";
+    const char *copyright = "";
+
+#if ((SDL_MIXER_MAJOR_VERSION >= 2) &&                                \
+     (SDL_MIXER_MAJOR_VERSION > 2 || SDL_MIXER_MINOR_VERSION >= 6) && \
+     (SDL_MIXER_MAJOR_VERSION > 2 || SDL_MIXER_MINOR_VERSION > 6 ||   \
+      SDL_MIXER_PATCHLEVEL >= 0))
+
+    title = Mix_GetMusicTitleTag(music);
+    album = Mix_GetMusicAlbumTag(music);
+    artist = Mix_GetMusicArtistTag(music);
+    copyright = Mix_GetMusicCopyrightTag(music);
+
+#endif
+
+    if (!music) {
+        return RAISE(pgExc_SDLError, "music not loaded");
+    }
+
+    meta_dict = Py_BuildValue("{ss ss ss ss}", "title", title, "album", album,
+                              "artist", artist, "copyright", copyright);
+
+    if (!current_music) {
+        Mix_FreeMusic(music);
+    }
```

Comment:
Won't this leak memory if there is music playing and the user still specifies a file to look at?

Reply:
Without freeing music it will certainly leak memory, so freeing it is necessary. If current music is playing it won't touch it, it will only free the loaded music, I could do a bit more testing though.

Reply:
I'm not sure you get what I'm saying.

Environment:
Music is playing (so current_music is not NULL)
User asks for pygame.mixer.music.get_metadata("another_file.mp3")

The `music` in get_metadata is only freed if current_music is NULL, so the function leaks memory when called in this configuration.